### DISCUSSION TITLE
Can't  login ssh connect to  remote  server (Wi-Fi  ) from termux, timeout

### DISCUSSION THREAD

**Discussion by opsyxc** [termux/termux-app#3314] [Q&A] [answered]
Hello Everyone , I have a problem 
Termux application version:0.118.0
Android OS version: Android 12
Device model: oneplus 7T
Centos version: 7

under mobile network  worked normally
but under Wi-Fi network
ssh -vvv  ip  -p22  
OpenSSH_9.3p1, OpenSSL 3.1.0 14 Mar 2023
debug1: Reading configuration data /data/data/com.termux/files/usr/etc/ssh/ssh_config
debug2: resolve_canonicalize: hostname ip is address
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts' -> '/data/data/com.termux/files/home/.ssh/known_hosts'
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts2' -> '/data/data/com.termux/files/home/.ssh/known_hosts2'
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug3: ssh_connect_direct: entering
debug1: Connecting to ip [ip] port 22.
debug3: set_sock_tos: set socket 3 IP_TOS 0x48
debug1: connect to addressip port 22: Connection timed out
ssh: connect to host ip port 22: Connection timed out

**Comment by sylirre** [comment, marked as answer]
> under mobile network worked normally
but under Wi-Fi network

Outgoing traffic firewall or ISP had to apply filter on outgoing SSH connections due to brute force attacks from his clients. If the latter, a connection over VPN should work.

Not an issue of `openssh` package since networking is basically same for Wifi and mobile.
